tools/template_based_query_generation/src/main/java/Utils.java

```diff
@@ -0,0 +1,185 @@
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * Utilities class that provides random and IO helper functions.
+ */
+public class Utils {
+
+	private static final ThreadLocalRandom random = ThreadLocalRandom.current();
+
+	private static final int lowerBound = 0;
+
+	private static final String CHARSET = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789_";
+
+	/**
+	 * Returns a random integer between a lowerBound and an upperBound, inclusive
+	 *
+	 * @param upperBound a non-negative integer upper bound on the generated random integer, inclusive
+	 * @return a random integer between lowerBound and upperBound, inclusive
+	 * @throws IllegalArgumentException if upperBound is negative
+	 */
+	public static int getRandomInteger(int upperBound) throws IllegalArgumentException {
+		if (upperBound < 0) {
+			throw new IllegalArgumentException("Upper bound cannot be negative");
+		}
+
+		return random.nextInt(lowerBound, upperBound + 1);
+	}
+
+	/**
+	 * Returns a random string with a specified length that matches the regex '[a-zA-Z_]'
+	 *
+	 * @param length a nonzero integer specifying the desired length of the generated string
+	 * @return a random string that matches the regex '[a-zA-Z_]' and has the specified length
+	 */
+	public static String getRandomString(int length) throws IllegalArgumentException {
+		if (length <= 0) {
+			throw new IllegalArgumentException("Random string must have positive length");
+		}
+
+		StringBuilder sb = new StringBuilder();
+
+		for (int i = 0; i < length; i++) {
+			int randomIndex = (int) (random.nextDouble() * CHARSET.length());
+			char randomChar = CHARSET.charAt(randomIndex);
+			if (i == 0 && Character.isDigit(randomChar)) {
+				// SQL identifiers can't start with digits, so replace with an arbitrary character
+				randomChar = 'a';
+			}
+			sb.append(randomChar);
+		}
+
+		return sb.toString();
+	}
+
+	/**
+	 * Writes generated outputs to a specified directory, creating one if it doesn't exist.
+	 *
+	 * @param outputs       collection of statements to write
+	 * @param directoryName relative path of a specified directory
+	 * @throws IOException if the IO fails or creating the necessary files or folders fails
+	 */
+	public static void writeDirectory(ImmutableMap<String, ImmutableList<String>> outputs, String directoryName) throws IOException {
+		String outputDirectory = getOutputDirectory(directoryName);
+		File file = new File(outputDirectory);
+
+		if (!file.exists() && !file.mkdir()) {
+			throw new FileNotFoundException("No such directory or the directory could not be created");
+		}
+
+		writeFile(outputs.get("BQ_skeletons"), outputDirectory + "/bq_skeleton.txt");
+		writeFile(outputs.get("BQ_tokenized"), outputDirectory + "/bq_tokenized.txt");
+		writeFile(outputs.get("Postgre_skeletons"), outputDirectory + "/postgre_skeleton.txt");
+		writeFile(outputs.get("Postgre_tokenized"), outputDirectory + "/postgre_tokenized.txt");
+		// TODO: write sample data to file
```

Comment:
Consider have assignment for each todo? For example "TODO(spoiledhua):..."

Reply:
Yup assigned to Victor